Attempting to add Darwin support for QGIS

**Describe the bug**
I'm attempting to add Darwin support for QGIS. As my first attempt, I've tried the naive path of changing https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/gis/qgis/unwrapped.nix#L54 from `platforms = with lib.platforms; linux;` to `platforms = with lib.platforms; all;` and running `nix-shell`. However, when building the dependencies, I run into this problem with the python module for Qscintilla:

```
Kairos:tmp ersin$ nix-shell -E "with import ../pkgs/top-level/impure.nix {}; libsForQt5.callPackage ../pkgs/applications/gis/qgis/unwrapped.nix { withGrass = false; }"
these derivations will be built:
  /nix/store/s9fx27g8p4m2q64wgk7lsrp44jsjka8i-python3.7-qscintilla-2.9.4.drv
  /nix/store/zdldyips2m3hwf1a2dc8dl9jh3llc17x-qtwebkit-5.212-alpha-01-26-2018.drv
these paths will be fetched (1.76 MiB download, 13.04 MiB unpacked):
  /nix/store/1y6wd1rk8v4krkhlx8cwsc6g27f0cy7d-qtwebchannel-5.11.3-dev
  /nix/store/ax02vbvxwjd3gwb7pqg34sjmvvq73iz3-qtsensors-5.11.3-dev
  /nix/store/c6pwax6zryik48z21lspxkmgfdfjhz8h-qtmultimedia-5.11.3-dev
  /nix/store/lm851ic3y8haisljnbkz0h0ffcflffrx-qtxmlpatterns-5.11.3-dev
  /nix/store/n3yknmr2sqlg0p3h53j44xgp5yhsjwv7-qttools-5.11.3-dev
  /nix/store/pgqklnbhzl0ykywv0zzwsn80wp9vg1by-qtdeclarative-5.11.3-dev
  /nix/store/xdkvl53b55zjm0k3g99cnbh17abrnvmn-qwt-6.1.2
  /nix/store/zcsyp52kc4dwnkw3sl8dmnpykmlllygh-qtlocation-5.11.3-dev
copying path '/nix/store/pgqklnbhzl0ykywv0zzwsn80wp9vg1by-qtdeclarative-5.11.3-dev' from 'https://cache.nixos.org'...
building '/nix/store/s9fx27g8p4m2q64wgk7lsrp44jsjka8i-python3.7-qscintilla-2.9.4.drv'...
Sourcing python-catch-conflicts-hook.sh
Sourcing python-remove-bin-bytecode-hook.sh
Sourcing python-imports-check-hook.sh
Using pythonImportsCheckPhase
unpacking sources
unpacking source archive /nix/store/7imkmyxwg710gx6dlj3n86rkdyw6yp8l-QScintilla_gpl-2.9.4.zip
source root is QScintilla_gpl-2.9.4
setting SOURCE_DATE_EPOCH to timestamp 1482706914 of file QScintilla_gpl-2.9.4/example-Qt4Qt5/mainwindow.h
patching sources
configuring
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/bin:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/lib:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/lib/python3.7:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/lib/python3.7/site-packages:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/lib/python3.7/site-packages/PyQt5-5.13.0.dist-info:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/lib/python3.7/site-packages/dbus:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/lib/python3.7/site-packages/dbus/mainloop:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/lib/python3.7/site-packages/PyQt5:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/lib/python3.7/site-packages/PyQt5/uic:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/lib/python3.7/site-packages/PyQt5/uic/port_v2:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/lib/python3.7/site-packages/PyQt5/uic/port_v3:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/lib/python3.7/site-packages/PyQt5/uic/widget-plugins:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/lib/python3.7/site-packages/PyQt5/uic/Loader:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/lib/python3.7/site-packages/PyQt5/uic/Compiler:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/share:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/share/sip:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/share/sip/PyQt5:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/share/sip/PyQt5/QtWebSockets:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/share/sip/PyQt5/QtPrintSupport:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/share/sip/PyQt5/QtOpenGL:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/share/sip/PyQt5/QtQml:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/share/sip/PyQt5/QtBluetooth:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/share/sip/PyQt5/QtNetwork:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/share/sip/PyQt5/QtDesigner:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/share/sip/PyQt5/QtMacExtras:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/share/sip/PyQt5/QtHelp:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/share/sip/PyQt5/Enginio:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/share/sip/PyQt5/QAxContainer:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/share/sip/PyQt5/QtSvg:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/share/sip/PyQt5/QtXml:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/share/sip/PyQt5/QtAndroidExtras:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/share/sip/PyQt5/QtRemoteObjects:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/share/sip/PyQt5/QtSensors:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/share/sip/PyQt5/QtWinExtras:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/share/sip/PyQt5/QtWebKitWidgets:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/share/sip/PyQt5/QtNetworkAuth:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/share/sip/PyQt5/QtGui:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/share/sip/PyQt5/QtNfc:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/share/sip/PyQt5/QtSerialPort:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/share/sip/PyQt5/QtQuick:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/share/sip/PyQt5/QtDBus:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/share/sip/PyQt5/QtX11Extras:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/share/sip/PyQt5/QtQuickWidgets:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/share/sip/PyQt5/QtWebChannel:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/share/sip/PyQt5/QtWidgets:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/share/sip/PyQt5/QtSql:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/share/sip/PyQt5/QtMultimedia:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/share/sip/PyQt5/QtLocation:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/share/sip/PyQt5/QtPositioning:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/share/sip/PyQt5/QtWebKit:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/share/sip/PyQt5/QtCore:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/share/sip/PyQt5/QtXmlPatterns:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/share/sip/PyQt5/QtMultimediaWidgets:
/nix/store/v2xkr37gffg5gfscjmm5dj2bbh9bqjh0-python3.7-pyqt-5.13.0/share/sip/PyQt5/QtTest:
Configuring QScintilla 2.9.4...
QScintilla 2.9.4 is being used.
The QScintilla .sip files will be installed in
/nix/store/n8yi5cg89zw27qaii17b40a4x796a6mv-python3.7-qscintilla-2.9.4/share/sip/PyQt5.
QScintilla will be installed in
/nix/store/n8yi5cg89zw27qaii17b40a4x796a6mv-python3.7-qscintilla-2.9.4/lib/python3.7/site-packages/PyQt5.
PyQt 5.13.0 is being used.
Qt 5.11.3 is being used.
sip 4.19.18 is being used.
The sip executable is
/nix/store/xqzavp71hz075bpnzz4v3amj2d2cvkah-python3.7-sip-4.19.18/bin/sip.
QScintilla is being built with 'protected' redefined as 'public'.
The PEP 484 stubs will be installed in
/nix/store/n8yi5cg89zw27qaii17b40a4x796a6mv-python3.7-qscintilla-2.9.4/lib/python3.7/site-packages/PyQt5.
The QScintilla API file will be installed in
/nix/store/n8yi5cg89zw27qaii17b40a4x796a6mv-python3.7-qscintilla-2.9.4/api/python3.7/api/python.
Generating the C++ source for the Qsci module...
Generating the .pro file for the Qsci module...
Generating the QScintilla API file...
Generating the top-level .pro file...
Generating the Makefiles...
no configure script, doing nothing
building
build flags: SHELL=/nix/store/9jd5ihjm10rwm3m6aj1wwbyx7f3cc902-bash-4.4-p23/bin/bash
cd Qsci/ && ( test -e Makefile || /nix/store/jfj2fmclsd1fmly02sfga7ddf42yvpaj-qtbase-5.11.3-dev/bin/qmake -o Makefile /private/tmp/nix-build-python3.7-qscintilla-2.9.4.drv-0/QScintilla_gpl-2.9.4/Python/Qsci/Qsci.pro ) && make -f Makefile
make[1]: Entering directory '/private/tmp/nix-build-python3.7-qscintilla-2.9.4.drv-0/QScintilla_gpl-2.9.4/Python/Qsci'
clang++ -c -pipe -stdlib=libc++ -fno-exceptions -O2 -Wall -W -fPIC -DSIP_PROTECTED_IS_PUBLIC -Dprotected=public -DQT_NO_DEBUG -DQT_PLUGIN -DQT_GUI_LIB -DQT_CORE_LIB -I. -isystem /nix/store/hrj338j7k2ghhbxvvrbb85ij65h876mh-qscintilla-qt5-2.9.4/include -I/nix/store/xqzavp71hz075bpnzz4v3amj2d2cvkah-python3.7-sip-4.19.18/include -isystem /nix/store/jz94a0lnq9p1i4wx1mrh2c9f1sppydq8-python3-3.7.4/include/python3.7m -isystem /nix/store/jfj2fmclsd1fmly02sfga7ddf42yvpaj-qtbase-5.11.3-dev/include/QtGui -isystem /nix/store/jfj2fmclsd1fmly02sfga7ddf42yvpaj-qtbase-5.11.3-dev/include/QtCore -I. -I/nix/store/srwv0hr8lcgdl8g0a67y6z8zs3xj1mf2-apple-framework-OpenGL/Library/Frameworks/OpenGL.framework/Headers -I/nix/store/40kgc5qczy3jpxznb3sxq7dzzmwv3dp4-apple-framework-AGL/Library/Frameworks/AGL.framework/Headers -I/nix/store/jfj2fmclsd1fmly02sfga7ddf42yvpaj-qtbase-5.11.3-dev/mkspecs/macx-clang -o sipQsciQsciLexerRuby.o sipQsciQsciLexerRuby.cpp
In file included from sip/qsciabstractapis.sip:24:
In file included from /nix/store/hrj338j7k2ghhbxvvrbb85ij65h876mh-qscintilla-qt5-2.9.4/include/Qsci/qsciabstractapis.h:33:
In file included from /nix/store/hrj338j7k2ghhbxvvrbb85ij65h876mh-qscintilla-qt5-2.9.4/include/Qsci/qsciscintilla.h:38:
In file included from /nix/store/hrj338j7k2ghhbxvvrbb85ij65h876mh-qscintilla-qt5-2.9.4/include/Qsci/qscicommand.h:31:
/nix/store/hrj338j7k2ghhbxvvrbb85ij65h876mh-qscintilla-qt5-2.9.4/include/Qsci/qsciscintillabase.h:30:10: fatal error: 'QAbstractScrollArea' file not found
#include <QAbstractScrollArea>
         ^~~~~~~~~~~~~~~~~~~~~
1 error generated.
make[1]: *** [Makefile:534: sipQsciQsciLexerRuby.o] Error 1
make[1]: Leaving directory '/private/tmp/nix-build-python3.7-qscintilla-2.9.4.drv-0/QScintilla_gpl-2.9.4/Python/Qsci'
make: *** [Makefile:45: sub-Qsci-make_first-ordered] Error 2
builder for '/nix/store/s9fx27g8p4m2q64wgk7lsrp44jsjka8i-python3.7-qscintilla-2.9.4.drv' failed with exit code 2
error: build of '/nix/store/5346sfiv2jfr14nnd8iznqilnydvv4hg-qtsensors-5.11.3.drv', '/nix/store/jbgqcan4rghm2s2y17vsxvh81w5j8yyq-qtxmlpatterns-5.11.3.drv', '/nix/store/klksr4awc1sbikiwhm964k6cz4vz0n0w-qwt-6.1.2.drv', '/nix/store/s9fx27g8p4m2q64wgk7lsrp44jsjka8i-python3.7-qscintilla-2.9.4.drv', '/nix/store/zdldyips2m3hwf1a2dc8dl9jh3llc17x-qtwebkit-5.212-alpha-01-26-2018.drv' failed
Kairos:tmp ersin$
```

I'm not sure how to satisfy the `QAbstractScrollArea` header file dependency. I did notice a file in qtbase in my store: `/nix/store/jfj2fmclsd1fmly02sfga7ddf42yvpaj-qtbase-5.11.3-dev/include/QtWidgets/QAbstractScrollArea`. My naive guess would be to pass `-I/nix/store/jfj2fmclsd1fmly02sfga7ddf42yvpaj-qtbase-5.11.3-dev/include/QtWidgets` to the compiler, but I don't know whether it's that trivial nor how to do this using Nix.

Any ideas?

**To Reproduce**
Steps to reproduce the behavior:
1. Clone Nixpkgs repo from master
2. `mkdir nixpkgs/tmp && cd nixpkgs/tmp`
3. `nix-shell -E "with import ../pkgs/top-level/impure.nix {}; libsForQt5.callPackage ../pkgs/applications/gis/qgis/unwrapped.nix { withGrass = false; }"` or alternatively `nix-shell -E "with import ../pkgs/top-level/impure.nix {}; callPackage ../pkgs/applications/gis/qgis "`

**Expected behavior**
I should be dropped into a `nix-shell` with all my dependencies built and available to begin hacking on QGIS.

**Screenshots**
N/A (see shell output above)

**Additional context**
Background: this is my first time modifying a Nix expression in Nixpkgs, indeed my first serious attempt at any kind of Nix packaging. I've been studying Nix very intently for the past few weeks and have gone through the Nix pills, the Nix manual, the Nixpkgs manual, and parts of Dostra's PhD thesis. I'm a beginner but not totally unfamiliar :-).

I'm also not too familiar with packaging Qt.

Finally, note that I'm trying to use my globally-installed Nix tools with my cloned nixpkgs. I'm not sure whether this is explicitly supported, please let me know if there's a better way.

**Metadata**
```
 - system: `"x86_64-darwin"`
 - host os: `Darwin 18.7.0, macOS 10.14.6`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.3`
 - channels(root): `"nixpkgs-20.03pre193309.c4196cca9ac"`
 - channels(ersin): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

Note that the above output is from running `nix run` globally, whereas I'm trying to use my local cloned copy of nixpkgs. (Perhaps this is a problem?)

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
  - qgis
  - qgis-unwrapped
  - qscintilla
```

--- a/pkgs/applications/gis/qgis/default.nix
+++ b/pkgs/applications/gis/qgis/default.nix
@@ -1,4 +1,6 @@
 {
+  lib,
+  stdenv,
   makeWrapper,
   nixosTests,
   symlinkJoin,
@@ -40,7 +42,8 @@ symlinkJoin {
         --prefix PATH : $program_PATH \
         --set PYTHONPATH $program_PYTHONPATH
     done
-
+  ''
+  + lib.optionalString (!stdenv.hostPlatform.isDarwin) ''
     ln -s ${qgis-unwrapped.man} $man
   '';
 

--- a/pkgs/applications/gis/qgis/ltr.nix
+++ b/pkgs/applications/gis/qgis/ltr.nix
@@ -1,4 +1,6 @@
 {
+  lib,
+  stdenv,
   makeWrapper,
   nixosTests,
   symlinkJoin,
@@ -14,15 +16,13 @@
 }:
 let
   qgis-ltr-unwrapped = libsForQt5.callPackage ./unwrapped-ltr.nix {
-    withGrass = withGrass;
-    withServer = withServer;
-    withWebKit = withWebKit;
+    inherit withGrass withServer withWebKit;
   };
 in
 
 symlinkJoin {
-  inherit (qgis-ltr-unwrapped) version src;
-  pname = "qgis";
+  inherit (qgis-ltr-unwrapped) version outputs src;
+  pname = "qgis-ltr";
 
   paths = [ qgis-ltr-unwrapped ];
 
@@ -43,6 +43,9 @@ symlinkJoin {
         --prefix PATH : $program_PATH \
         --set PYTHONPATH $program_PYTHONPATH
     done
+  ''
+  + lib.optionalString (!stdenv.hostPlatform.isDarwin) ''
+    ln -s ${qgis-ltr-unwrapped.man} $man
   '';
 
   passthru = {

--- a/pkgs/applications/gis/qgis/unwrapped-ltr.nix
+++ b/pkgs/applications/gis/qgis/unwrapped-ltr.nix
@@ -1,6 +1,8 @@
 {
   lib,
+  stdenv,
   fetchFromGitHub,
+  lndir,
   makeWrapper,
   mkDerivation,
   replaceVars,
@@ -11,6 +13,8 @@
   withServer,
   withWebKit,
 
+  darwin,
+  libtasn1,
   bison,
   cmake,
   draco,
@@ -40,6 +44,7 @@
   qtmultimedia,
   qtsensors,
   qtserialport,
+  qtsvg,
   qtwebkit,
   qtxmlpatterns,
   qwt,
@@ -68,6 +73,7 @@ let
     psycopg2
     pygments
     pyqt5
+    pyqt5-sip
     pyqt-builder
     python-dateutil
     pytz
@@ -83,6 +89,7 @@ in
 mkDerivation rec {
   version = "3.40.14";
   pname = "qgis-ltr-unwrapped";
+  outputs = [ "out" ] ++ lib.optional (!stdenv.hostPlatform.isDarwin) "man";
 
   src = fetchFromGitHub {
     owner = "qgis";
@@ -105,6 +112,10 @@ mkDerivation rec {
     cmake
     flex
     ninja
+  ]
+  ++ lib.optionals stdenv.hostPlatform.isDarwin [
+    darwin.autoSignDarwinBinariesHook
+    lndir
   ];
 
   buildInputs = [
@@ -131,6 +142,7 @@ mkDerivation rec {
     qtmultimedia
     qtsensors
     qtserialport
+    qtsvg
     qtxmlpatterns
     qwt
     sqlite
@@ -139,6 +151,7 @@ mkDerivation rec {
   ]
   ++ lib.optional withGrass grass
   ++ lib.optional withWebKit qtwebkit
+  ++ lib.optional stdenv.hostPlatform.isDarwin libtasn1
   ++ pythonBuildInputs;
 
   patches = [
@@ -165,6 +178,10 @@ mkDerivation rec {
     # See https://github.com/libspatialindex/libspatialindex/issues/276
     "-DWITH_INTERNAL_SPATIALINDEX=True"
   ]
+  ++ lib.optionals stdenv.hostPlatform.isDarwin [
+    "-DQGIS_MACAPP_BUNDLE=0" # Don't copy Qt into bundle; we fix paths in postFixup
+    "-DSQLITE3_INCLUDE_DIR=${sqlite.dev}/include" # FindSqlite3.cmake incorrectly assumes framework
+  ]
   ++ lib.optional (!withWebKit) "-DWITH_QTWEBKIT=OFF"
   ++ lib.optional withServer [
     "-DWITH_SERVER=True"
@@ -183,18 +200,102 @@ mkDerivation rec {
   ];
 
   dontWrapGApps = true; # wrapper params passed below
+  dontWrapQtApps = stdenv.hostPlatform.isDarwin;
 
-  postFixup = lib.optionalString withGrass ''
-    # GRASS has to be availble on the command line even though we baked in
-    # the path at build time using GRASS_PREFIX.
-    # Using wrapGAppsHook also prevents file dialogs from crashing the program
-    # on non-NixOS.
-    for program in $out/bin/*; do
-      wrapProgram $program \
-        "''${gappsWrapperArgs[@]}" \
-        --prefix PATH : ${lib.makeBinPath [ grass ]}
-    done
-  '';
+  postFixup =
+    lib.optionalString (withGrass && stdenv.hostPlatform.isLinux) ''
+      # GRASS has to be available on the command line even though we baked in
+      # the path at build time using GRASS_PREFIX.
+      # Using wrapGAppsHook also prevents file dialogs from crashing the program
+      # on non-NixOS.
+      for program in $out/bin/*; do
+        wrapProgram $program \
+          "''${gappsWrapperArgs[@]}" \
+          --prefix PATH : ${lib.makeBinPath [ grass ]}
+      done
+    ''
+    + lib.optionalString stdenv.hostPlatform.isDarwin ''
+            mkdir -p $out/Applications $out/bin
+            mv $out/QGIS.app $out/Applications/
+            ln -s $out/Applications/QGIS.app/Contents/MacOS/QGIS $out/bin/qgis
+
+            SHORT_VERSION=$(echo "${version}" | cut -d. -f1,2)
+            BUNDLE="$out/Applications/QGIS.app"
+            FRAMEWORKS="$BUNDLE/Contents/Frameworks"
+
+            fix_binary() {
+              local f="$1"
+              [[ -f "$f" ]] || return 0
+              file "$f" | grep -q "Mach-O" || return 0
+
+              install_name_tool \
+                -change "@loader_path/../lib/libqscintilla2_qt5.dylib" "${qscintilla}/lib/libqscintilla2_qt5.dylib" \
+                -change "@loader_path/../lib/libqt5keychain.dylib" "${qtkeychain}/lib/libqt5keychain.dylib" \
+                -change "@loader_path/../lib/libqwt.dylib" "${qwt}/lib/libqwt.dylib" \
+                -change "@loader_path/../../Frameworks/qca-qt5.framework/qca-qt5" "${qca-qt5}/lib/qca-qt5.framework/qca-qt5" \
+                -change "@loader_path/../../../qca-qt5.framework/qca-qt5" "${qca-qt5}/lib/qca-qt5.framework/qca-qt5" \
+                -change "@loader_path/../../../../MacOS/lib/libqscintilla2_qt5.dylib" "${qscintilla}/lib/libqscintilla2_qt5.dylib" \
+                -change "@loader_path/../../../../MacOS/lib/libqt5keychain.dylib" "${qtkeychain}/lib/libqt5keychain.dylib" \
+                -change "@loader_path/../../../../MacOS/lib/libqwt.dylib" "${qwt}/lib/libqwt.dylib" \
+                -change "@executable_path/lib/libqwt.dylib" "${qwt}/lib/libqwt.dylib" \
+                -change "@executable_path/lib/libqscintilla2_qt5.dylib" "${qscintilla}/lib/libqscintilla2_qt5.dylib" \
+                -change "@executable_path/lib/libqt5keychain.dylib" "${qtkeychain}/lib/libqt5keychain.dylib" \
+                -change "@executable_path/../Frameworks/qca-qt5.framework/qca-qt5" "${qca-qt5}/lib/qca-qt5.framework/qca-qt5" \
+                -change "@loader_path/../../../qgis_core.framework/qgis_core" "$FRAMEWORKS/qgis_core.framework/Versions/$SHORT_VERSION/qgis_core" \
+                -change "@loader_path/../../../qgis_gui.framework/qgis_gui" "$FRAMEWORKS/qgis_gui.framework/Versions/$SHORT_VERSION/qgis_gui" \
+                -change "@loader_path/../../../qgis_analysis.framework/qgis_analysis" "$FRAMEWORKS/qgis_analysis.framework/Versions/$SHORT_VERSION/qgis_analysis" \
+                -change "@loader_path/../../../qgis_3d.framework/qgis_3d" "$FRAMEWORKS/qgis_3d.framework/Versions/$SHORT_VERSION/qgis_3d" \
+                -change "@loader_path/../../../qgis_native.framework/qgis_native" "$FRAMEWORKS/qgis_native.framework/Versions/$SHORT_VERSION/qgis_native" \
+                "$f" 2>/dev/null || true
+            }
+
+            fix_binary "$BUNDLE/Contents/MacOS/QGIS"
+            for lib in "$BUNDLE/Contents/MacOS/lib"/*.dylib; do fix_binary "$lib"; done
+            for fw in qgis_core qgis_gui qgis_analysis qgis_3d qgis_native; do
+              fix_binary "$FRAMEWORKS/$fw.framework/Versions/$SHORT_VERSION/$fw"
+              [[ -f "$FRAMEWORKS/$fw.framework/$fw" && ! -L "$FRAMEWORKS/$fw.framework/$fw" ]] && \
+                fix_binary "$FRAMEWORKS/$fw.framework/$fw"
+            done
+            for plugin in "$BUNDLE/Contents/PlugIns/qgis"/*.so; do fix_binary "$plugin"; done
+
+            ${lib.optionalString withGrass ''
+              fix_binary "$FRAMEWORKS/qgisgrass8.framework/Versions/$SHORT_VERSION/qgisgrass8"
+              install_name_tool \
+                -change "@loader_path/../../../qgisgrass8.framework/qgisgrass8" "$FRAMEWORKS/qgisgrass8.framework/Versions/$SHORT_VERSION/qgisgrass8" \
+                "$BUNDLE/Contents/MacOS/QGIS" 2>/dev/null || true
+              for lib in "$BUNDLE/Contents/MacOS/lib"/*.dylib; do
+                install_name_tool \
+                  -change "@loader_path/../../../qgisgrass8.framework/qgisgrass8" "$FRAMEWORKS/qgisgrass8.framework/Versions/$SHORT_VERSION/qgisgrass8" \
+                  "$lib" 2>/dev/null || true
+              done
+            ''}
+
+            ${lib.optionalString withServer ''
+              fix_binary "$BUNDLE/Contents/MacOS/lib/libqgis_server.${version}.dylib"
+            ''}
+
+            # Merge Python packages (lndir handles namespace packages correctly)
+            QGIS_PYTHON="$BUNDLE/Contents/Resources/python"
+            for pkg in ${
+              lib.concatMapStringsSep " " (p: "${p}/${py.pkgs.python.sitePackages}") (
+                py.pkgs.requiredPythonModules pythonBuildInputs
+              )
+            }; do
+              [[ -d "$pkg" ]] && lndir -silent "$pkg" "$QGIS_PYTHON"
+            done
+            # Remove broken symlinks
+            find "$QGIS_PYTHON" -type l ! -exec test -e {} \; -delete
+
+            # Create merged Qt plugins directory in the bundle (LSEnvironment is unreliable)
+            mkdir -p "$BUNDLE/Contents/PlugIns"
+            lndir -silent "${qtbase}/${qtbase.qtPluginPrefix}" "$BUNDLE/Contents/PlugIns"
+            lndir -silent "${qtsvg.bin}/${qtbase.qtPluginPrefix}" "$BUNDLE/Contents/PlugIns"
+
+            cat > "$BUNDLE/Contents/Resources/qt.conf" << 'EOF'
+      [Paths]
+      Plugins = PlugIns
+      EOF
+    '';
 
   # >9k objects, >3h build time on a normal build slot
   requiredSystemFeatures = [ "big-parallel" ];
@@ -204,6 +305,6 @@ mkDerivation rec {
     homepage = "https://www.qgis.org";
     license = lib.licenses.gpl2Plus;
     teams = [ lib.teams.geospatial ];
-    platforms = with lib.platforms; linux;
+    platforms = lib.platforms.unix;
   };
 }

--- a/pkgs/applications/gis/qgis/unwrapped.nix
+++ b/pkgs/applications/gis/qgis/unwrapped.nix
@@ -1,6 +1,8 @@
 {
   lib,
+  stdenv,
   fetchFromGitHub,
+  lndir,
   makeWrapper,
   mkDerivation,
   replaceVars,
@@ -11,6 +13,8 @@
   withServer,
   withWebKit,
 
+  darwin,
+  libtasn1,
   bison,
   cmake,
   draco,
@@ -40,6 +44,7 @@
   qtmultimedia,
   qtsensors,
   qtserialport,
+  qtsvg,
   qtwebkit,
   qtxmlpatterns,
   qwt,
@@ -68,6 +73,7 @@ let
     psycopg2
     pygments
     pyqt5
+    pyqt5-sip
     pyqt-builder
     python-dateutil
     pytz
@@ -83,10 +89,7 @@ in
 mkDerivation rec {
   version = "3.44.6";
   pname = "qgis-unwrapped";
-  outputs = [
-    "out"
-    "man"
-  ];
+  outputs = [ "out" ] ++ lib.optional (!stdenv.hostPlatform.isDarwin) "man";
 
   src = fetchFromGitHub {
     owner = "qgis";
@@ -109,6 +112,10 @@ mkDerivation rec {
     cmake
     flex
     ninja
+  ]
+  ++ lib.optionals stdenv.hostPlatform.isDarwin [
+    darwin.autoSignDarwinBinariesHook
+    lndir
   ];
 
   buildInputs = [
@@ -135,6 +142,7 @@ mkDerivation rec {
     qtmultimedia
     qtsensors
     qtserialport
+    qtsvg
     qtxmlpatterns
     qwt
     sqlite
@@ -143,6 +151,7 @@ mkDerivation rec {
   ]
   ++ lib.optional withGrass grass
   ++ lib.optional withWebKit qtwebkit
+  ++ lib.optional stdenv.hostPlatform.isDarwin libtasn1
   ++ pythonBuildInputs;
 
   patches = [
@@ -165,6 +174,10 @@ mkDerivation rec {
     # See https://github.com/libspatialindex/libspatialindex/issues/276
     "-DWITH_INTERNAL_SPATIALINDEX=True"
   ]
+  ++ lib.optionals stdenv.hostPlatform.isDarwin [
+    "-DQGIS_MACAPP_BUNDLE=0" # Don't copy Qt into bundle; we fix paths in postFixup
+    "-DSQLITE3_INCLUDE_DIR=${sqlite.dev}/include" # FindSqlite3.cmake incorrectly assumes framework
+  ]
   ++ lib.optional (!withWebKit) "-DWITH_QTWEBKIT=OFF"
   ++ lib.optional withServer [
     "-DWITH_SERVER=True"
@@ -183,18 +196,102 @@ mkDerivation rec {
   ];
 
   dontWrapGApps = true; # wrapper params passed below
+  dontWrapQtApps = stdenv.hostPlatform.isDarwin;
 
-  postFixup = lib.optionalString withGrass ''
-    # GRASS has to be available on the command line even though we baked in
-    # the path at build time using GRASS_PREFIX.
-    # Using wrapGAppsHook also prevents file dialogs from crashing the program
-    # on non-NixOS.
-    for program in $out/bin/*; do
-      wrapProgram $program \
-        "''${gappsWrapperArgs[@]}" \
-        --prefix PATH : ${lib.makeBinPath [ grass ]}
-    done
-  '';
+  postFixup =
+    lib.optionalString (withGrass && stdenv.hostPlatform.isLinux) ''
+      # GRASS has to be available on the command line even though we baked in
+      # the path at build time using GRASS_PREFIX.
+      # Using wrapGAppsHook also prevents file dialogs from crashing the program
+      # on non-NixOS.
+      for program in $out/bin/*; do
+        wrapProgram $program \
+          "''${gappsWrapperArgs[@]}" \
+          --prefix PATH : ${lib.makeBinPath [ grass ]}
+      done
+    ''
+    + lib.optionalString stdenv.hostPlatform.isDarwin ''
+            mkdir -p $out/Applications $out/bin
+            mv $out/QGIS.app $out/Applications/
+            ln -s $out/Applications/QGIS.app/Contents/MacOS/QGIS $out/bin/qgis
+
+            SHORT_VERSION=$(echo "${version}" | cut -d. -f1,2)
+            BUNDLE="$out/Applications/QGIS.app"
+            FRAMEWORKS="$BUNDLE/Contents/Frameworks"
+
+            fix_binary() {
+              local f="$1"
+              [[ -f "$f" ]] || return 0
+              file "$f" | grep -q "Mach-O" || return 0
+
+              install_name_tool \
+                -change "@loader_path/../lib/libqscintilla2_qt5.dylib" "${qscintilla}/lib/libqscintilla2_qt5.dylib" \
+                -change "@loader_path/../lib/libqt5keychain.dylib" "${qtkeychain}/lib/libqt5keychain.dylib" \
+                -change "@loader_path/../lib/libqwt.dylib" "${qwt}/lib/libqwt.dylib" \
+                -change "@loader_path/../../Frameworks/qca-qt5.framework/qca-qt5" "${qca-qt5}/lib/qca-qt5.framework/qca-qt5" \
+                -change "@loader_path/../../../qca-qt5.framework/qca-qt5" "${qca-qt5}/lib/qca-qt5.framework/qca-qt5" \
+                -change "@loader_path/../../../../MacOS/lib/libqscintilla2_qt5.dylib" "${qscintilla}/lib/libqscintilla2_qt5.dylib" \
+                -change "@loader_path/../../../../MacOS/lib/libqt5keychain.dylib" "${qtkeychain}/lib/libqt5keychain.dylib" \
+                -change "@loader_path/../../../../MacOS/lib/libqwt.dylib" "${qwt}/lib/libqwt.dylib" \
+                -change "@executable_path/lib/libqwt.dylib" "${qwt}/lib/libqwt.dylib" \
+                -change "@executable_path/lib/libqscintilla2_qt5.dylib" "${qscintilla}/lib/libqscintilla2_qt5.dylib" \
+                -change "@executable_path/lib/libqt5keychain.dylib" "${qtkeychain}/lib/libqt5keychain.dylib" \
+                -change "@executable_path/../Frameworks/qca-qt5.framework/qca-qt5" "${qca-qt5}/lib/qca-qt5.framework/qca-qt5" \
+                -change "@loader_path/../../../qgis_core.framework/qgis_core" "$FRAMEWORKS/qgis_core.framework/Versions/$SHORT_VERSION/qgis_core" \
+                -change "@loader_path/../../../qgis_gui.framework/qgis_gui" "$FRAMEWORKS/qgis_gui.framework/Versions/$SHORT_VERSION/qgis_gui" \
+                -change "@loader_path/../../../qgis_analysis.framework/qgis_analysis" "$FRAMEWORKS/qgis_analysis.framework/Versions/$SHORT_VERSION/qgis_analysis" \
+                -change "@loader_path/../../../qgis_3d.framework/qgis_3d" "$FRAMEWORKS/qgis_3d.framework/Versions/$SHORT_VERSION/qgis_3d" \
+                -change "@loader_path/../../../qgis_native.framework/qgis_native" "$FRAMEWORKS/qgis_native.framework/Versions/$SHORT_VERSION/qgis_native" \
+                "$f" 2>/dev/null || true
+            }
+
+            fix_binary "$BUNDLE/Contents/MacOS/QGIS"
+            for lib in "$BUNDLE/Contents/MacOS/lib"/*.dylib; do fix_binary "$lib"; done
+            for fw in qgis_core qgis_gui qgis_analysis qgis_3d qgis_native; do
+              fix_binary "$FRAMEWORKS/$fw.framework/Versions/$SHORT_VERSION/$fw"
+              [[ -f "$FRAMEWORKS/$fw.framework/$fw" && ! -L "$FRAMEWORKS/$fw.framework/$fw" ]] && \
+                fix_binary "$FRAMEWORKS/$fw.framework/$fw"
+            done
+            for plugin in "$BUNDLE/Contents/PlugIns/qgis"/*.so; do fix_binary "$plugin"; done
+
+            ${lib.optionalString withGrass ''
+              fix_binary "$FRAMEWORKS/qgisgrass8.framework/Versions/$SHORT_VERSION/qgisgrass8"
+              install_name_tool \
+                -change "@loader_path/../../../qgisgrass8.framework/qgisgrass8" "$FRAMEWORKS/qgisgrass8.framework/Versions/$SHORT_VERSION/qgisgrass8" \
+                "$BUNDLE/Contents/MacOS/QGIS" 2>/dev/null || true
+              for lib in "$BUNDLE/Contents/MacOS/lib"/*.dylib; do
+                install_name_tool \
+                  -change "@loader_path/../../../qgisgrass8.framework/qgisgrass8" "$FRAMEWORKS/qgisgrass8.framework/Versions/$SHORT_VERSION/qgisgrass8" \
+                  "$lib" 2>/dev/null || true
+              done
+            ''}
+
+            ${lib.optionalString withServer ''
+              fix_binary "$BUNDLE/Contents/MacOS/lib/libqgis_server.${version}.dylib"
+            ''}
+
+            # Merge Python packages (lndir handles namespace packages correctly)
+            QGIS_PYTHON="$BUNDLE/Contents/Resources/python"
+            for pkg in ${
+              lib.concatMapStringsSep " " (p: "${p}/${py.pkgs.python.sitePackages}") (
+                py.pkgs.requiredPythonModules pythonBuildInputs
+              )
+            }; do
+              [[ -d "$pkg" ]] && lndir -silent "$pkg" "$QGIS_PYTHON"
+            done
+            # Remove broken symlinks
+            find "$QGIS_PYTHON" -type l ! -exec test -e {} \; -delete
+
+            # Create merged Qt plugins directory in the bundle (LSEnvironment is unreliable)
+            mkdir -p "$BUNDLE/Contents/PlugIns"
+            lndir -silent "${qtbase}/${qtbase.qtPluginPrefix}" "$BUNDLE/Contents/PlugIns"
+            lndir -silent "${qtsvg.bin}/${qtbase.qtPluginPrefix}" "$BUNDLE/Contents/PlugIns"
+
+            cat > "$BUNDLE/Contents/Resources/qt.conf" << 'EOF'
+      [Paths]
+      Plugins = PlugIns
+      EOF
+    '';
 
   # >9k objects, >3h build time on a normal build slot
   requiredSystemFeatures = [ "big-parallel" ];
@@ -204,6 +301,6 @@ mkDerivation rec {
     homepage = "https://www.qgis.org";
     license = lib.licenses.gpl2Plus;
     teams = [ lib.teams.geospatial ];
-    platforms = with lib.platforms; linux;
+    platforms = lib.platforms.unix;
   };
 }

